test/jdk/javax/management/Introspector/ClassLeakTest.java

```diff
@@ -48,38 +49,10 @@ public static void main(String[] args) throws Exception {
                            "Standard MBean does not retain a reference to " +
                            "the MBean's class");
 
-
-        String[] cpaths = System.getProperty("test.classes", ".")
-                                .split(File.pathSeparator);
-        URL[] urls = new URL[cpaths.length];
-        for (int i=0; i < cpaths.length; i++) {
-            urls[i] = Paths.get(cpaths[i]).toUri().toURL();
-        }
-
-        @SuppressWarnings("removal")
-        PrivateMLet mlet = new PrivateMLet(urls, null, false);
-        Class<?> shadowClass = mlet.loadClass(TestMBean.class.getName());
-        if (shadowClass == TestMBean.class) {
-            System.out.println("TEST INVALID: MLet got original " +
-                               "TestMBean not shadow");
-            System.exit(1);
-        }
-        shadowClass = null;
-
         MBeanServer mbs = MBeanServerFactory.createMBeanServer();
-        ObjectName mletName = new ObjectName("x:type=mlet");
-        mbs.registerMBean(mlet, mletName);
-
-        ObjectName testName = new ObjectName("x:type=test");
-        mbs.createMBean(Test.class.getName(), testName, mletName);
-
-        ClassLoader testLoader = mbs.getClassLoaderFor(testName);
-        if (testLoader != mlet) {
-            System.out.println("TEST INVALID: MBean's class loader is not " +
-                               "MLet: " + testLoader);
-            System.exit(1);
-        }
-        testLoader = null;
+        ObjectName testName = new ObjectName("x:name=Test");
```

Comment:
Q: I'd just like to understand why it is `x:name=Test` instead of `x:type=Test` as it was before.

Reply:
These ObjectName patterns (domain:key=value) don't matter very much for these test mbeans.
We just need an ObjectName to pass when registering the MBean, and also to use when querying, e.g. calling getMBeanInfo.

The domain part was "x" but could be empty, but it must also define at least one key=value.
The "type=mlet" might still work (I just tested: it does 8-) ), but I wanted to remove it as it would be misleading.
So we need to add some key=value and just giving a name is enough.

Reply:
In ClassLeakTest.java, it was creating two MBeans, but now only creates one.
It was creating a PrivateMLet (which it registers using type=mlet) and comparing classloader with another MBean.  The second MBean is created using the m-let name as the "loader name" parameter to mbs.createMBean().  This doesn't make sense without MLets, which are classloaders.  

The test retains is stated goal of checking a reference to the MBean class is not retained after it is unloaded.